File: spring-webflux/src/main/java/software/darkmatter/school/blog/config/security/SimpleAuthenticationManager.java
package software.darkmatter.school.blog.config.security;

import org.springframework.security.authentication.ReactiveAuthenticationManager;
import org.springframework.security.core.Authentication;
import org.springframework.stereotype.Component;
import reactor.core.publisher.Mono;

@Component
public class SimpleAuthenticationManager implements ReactiveAuthenticationManager {

    @Override
    public Mono<Authentication> authenticate(Authentication authentication) {
        if (SimpleAuthentication.class.isAssignableFrom(authentication.getClass())) {
            return Mono.just(authentication);
        }
        return Mono.empty();
    }
}

File: kotlin-spring-webflux/src/main/kotlin/software/darkmatter/school/blog/config/security/SimpleAuthenticationManager.kt
package software.darkmatter.school.blog.config.security

import org.springframework.security.authentication.ReactiveAuthenticationManager
import org.springframework.security.core.Authentication
import org.springframework.stereotype.Component
import reactor.core.publisher.Mono

@Component
class SimpleAuthenticationManager : ReactiveAuthenticationManager {

    override fun authenticate(authentication: Authentication): Mono<Authentication> {
        return when (authentication) {
            is SimpleAuthentication -> {
                Mono.just(authentication)
            }
            else -> Mono.empty()
        }
    }
}
